includes/class-wp-job-manager-rest-api.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Handles functionality related to the REST API.
+ *
+ * @package wp-job-manager
+ * @since 1.33.0
+ */
+class WP_Job_Manager_REST_API {
+	/**
+	 * Sets up initial hooks.
+	 *
+	 * @static
+	 */
+	public static function init() {
+		add_action( 'rest_prepare_job_listing', array( __CLASS__, 'prepare_job_listing' ), 10, 2 );
+	}
+
+	/**
+	 * Filters the job listing data for a REST API response.
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param WP_Post          $post     Post object.
+	 * @return WP_REST_Response
+	 */
+	public static function prepare_job_listing( $response, $post ) {
+		$current_user = wp_get_current_user();
+		$fields       = WP_Job_Manager_Post_Types::get_job_listing_fields();
+		$data         = $response->get_data();
+
+		foreach ( $data['meta'] as $meta_key => $meta_value ) {
+			if ( isset( $fields[ $meta_key ] ) && is_callable( $fields[ $meta_key ]['auth_view_callback'] ) ) {
```

Comment:
Maybe also check for `isset( $fields[ $meta_key ]['auth_view_callback'] )`

Reply:
I've set defaults so it should always be set. Sometimes it could be `null`, but that shouldn't cause a problem with this.